core/src/main/java/org/infinispan/stream/impl/IntermediateType.java

```diff
@@ -0,0 +1,61 @@
+package org.infinispan.stream.impl;
+
+import org.infinispan.CacheStream;
+import org.infinispan.CacheStreamable;
+import org.infinispan.DoubleCacheStream;
+import org.infinispan.IntCacheStream;
+import org.infinispan.LongCacheStream;
+
+import java.util.Arrays;
+import java.util.Spliterator;
+import java.util.stream.BaseStream;
+import java.util.stream.IntStream;
+import java.util.stream.LongStream;
+import java.util.stream.Stream;
+import java.util.stream.StreamSupport;
+
+/**
+ * The various intermediate types.  Note that the local intermediate operation can be a type other than the
+ * resulting stream, so we have to keep track of what it was to avoid {@link ClassCastException} issues.
+ * <p>
+ * Currently the streams generated vary based on if it wants an iterator.  This should hopefully be faster, as an
+ * iterator can require many messages, however it requires a bit more memory.  Thus terminal operations that can
```

Comment:
I apologize, this was a remnant from when I was either performing an iterator operation like now or for terminal operations that require the entire stream just calling toArray.  I can remove this section.